ivy/functional/frontends/numpy/statistics/order_statistics.py

```diff
@@ -0,0 +1,107 @@
+# global
+import numpy as np
+from typing import Optional, Sequence, Union
+import ivy
+
+
+def _quantile_is_valid(q):
+    # avoid expensive reductions, relevant for arrays with < O(1000) elements
+    if q.ndim == 1 and q.size < 10:
+        for i in range(q.size):
+            if not (0.0 <= q[i] <= 1.0):
+                return False
+    else:
+        if not (ivy.all(0 <= q) and ivy.all(q <= 1)):
+            return False
+    return True
+
+
+
+def cpercentile(N, percent, key=lambda x: x):
+    """
+    Find the percentile   of a list of values.
+
+    @parameter N - is a list of values. Note N MUST BE already sorted.
+    @parameter percent - a float value from 0.0 to 1.0.
+    @parameter key - optional key function to compute value from each element of N.
+
+    @return - the percentile  of the values
+    """
+    N.sort()
+    k = (len(N) - 1) * percent
+    f = ivy.math.floor(k)
+    c = ivy.math.ceil(k)
+    if f == c:
+        return key(N[int(k)])
+    d0 = key(N[int(f)]) * (c - k)
+    d1 = key(N[int(c)]) * (k - f)
+    return d0 + d1

```

Comment:
We should this make function private, this function is basically a helper function for `nanpercentile`, it shouldn't be allowed to be accessed publicly.

Reply:
Done!